tracer/src/Datadog.Trace/AppSec/ControllerContextExtensions.Framework.cs

```diff
@@ -28,43 +28,55 @@ internal static void MonitorBodyAndPathParams(this IControllerContext controller
             var context = HttpContext.Current;
             var iastEnabled = Iast.Iast.Instance.Settings.Enabled;
             Scope scope = null;
+            object requestBody = null;
+            Dictionary<string, object> pathParamsDic = null;
 
-            if ((context != null && security.Enabled) || iastEnabled)
+            if (context != null)
             {
-                scope = SharedItems.TryPeekScope(context, peekScopeKey);
-            }
-
-            if (context != null && security.Enabled)
-            {
-                var bodyDic = new Dictionary<string, object>(parameters.Count);
-                var pathParamsDic = new Dictionary<string, object>(parameters.Count);
-                foreach (var item in parameters)
+                if (security.Enabled || iastEnabled)
                 {
-                    if (controllerContext.RouteData.Values.ContainsKey(item.Key))
-                    {
-                        pathParamsDic[item.Key] = item.Value;
-                    }
-                    else
+                    scope = SharedItems.TryPeekScope(context, peekScopeKey);
+
+                    var bodyDic = new Dictionary<string, object>(parameters.Count);
+                    pathParamsDic = new Dictionary<string, object>(parameters.Count);
+                    foreach (var item in parameters)
                     {
-                        bodyDic[item.Key] = item.Value;
+                        if (controllerContext.RouteData.Values.ContainsKey(item.Key))
+                        {
+                            pathParamsDic[item.Key] = item.Value;
+                        }
+                        else
+                        {
+                            // We exclude the query string params
+                            if (!context.Request.QueryString.AllKeys.Contains(item.Key))
```

Comment:
Just to confirm, is this something you want to do for _both_ security _and_ IAST?

Reply:
Yes, The bodyDic should not contain query string params in any case.